### ISSUE TITLE
connecting to a secure MongoDB installation

### ISSUE THREAD

**Issue by goldiiman** [Studio3T/robomongo#1357]
I have a mongo installation where the user has to authenticate against
Admin but the user is locked down to a different database.

as an example, they cannot use "show dbs" but if I use Mongo shell the below works

mongo <ip>:<port>/admin -u <user> -p PWD
show dbs <-- this would fail
use <database>
show collections
db.<collection>find({})

in Robotmongo I have set DefaultDatabase to the correct one but it shows admin in the tree, how can I switch the tree to the same database set in the DefaultDatabase?

**Comment by simsekgokhan** [comment]
@goldiiman , thanks for reporting the problem.
Can you share versions of your OS, Robomongo and mongodb please?

As I understand you are talking about "admin" database seen in the left pane tree of your active connection. But that's not the default database, it is just admin database. The default database for Robomongo is the database which Robomongo shell connects to by default when open shell is not triggered from a specific database. (i.e. right click on your connection->Open Shell operation will connect to the default database specified in your Robomongo connection settings -> Advanced tab -> Default Database)

If my understanding is not correct, can you share a screenshot of the problem or the desired behavior you want to achieve, please? It will be much easier to understand your problem.

**Comment by goldiiman** [comment]
hi

I am using a Mac

the mongo script below works fine (connection details provided). please try
it on your GUI


mongo 188.166.145.148:63487/admin -u user1ncl -p 7878

use A30B82DC0CA947C98C8A2424A28AA6A37AB53BF943BAAFEC7D5E741B7232AC6

show collections

db.chatbotcomparisionsxlsx.find({})



thanks
Martin

**Comment by vandy125** [comment]
I believe that I have the same issue described up above.  User is locked down to a different database than is authenticated against.  In my case, user can connect through command line and see collections on a different database using a command like this:
mongo server.location.com/WidgetDatabase -u userAccount -p userPassword --authenticationDatabase admin

show collections and querying the collections returns data from the command line in the WidgetDatabase.
show dbs fails (Not authorized on admin to execute command)

I've tried Robo Mongo versions 0.9.0, 1.0.0, and 1.1.0.  The database version is 3.2.4, and RoboMongo is running on Windows Server 2012.

Attached is a screenshot of what RoboMongo looks like as well as the connection settings that I have tried.  It is missing all of the collections.

![robomongoadvancedsettings](https://user-images.githubusercontent.com/29871573/27805299-83cc167a-5ff9-11e7-8504-7554a0091dd3.JPG)
![robomongoauthenticationsettings](https://user-images.githubusercontent.com/29871573/27805302-83d1fb58-5ff9-11e7-9f6c-68df61eeba3d.JPG)
![robomongoconnectionsettings](https://user-images.githubusercontent.com/29871573/27805300-83cd13d6-5ff9-11e7-83ee-85d12f3a31e8.JPG)
![robomongocollectionview](https://user-images.githubusercontent.com/29871573/27805301-83cf2c84-5ff9-11e7-9dd9-21f246b7396c.JPG)

**Comment by goldiiman** [comment]
In the RoboMongo GUI I can goto the shell and switch to the Database but
ideally I would have expected the tree view to show the Default database I
had entered and its collections

**Comment by ctmay4** [comment]
We just added authentication to a database and now see the exact same behavior described above.  Able to "use myDb" in the shell, but unable to see them in the tree view.  Really makes Robomongo much less useful if you can't see that view. 

Any thoughts on this?

**Comment by ctmay4** [comment]
Did some more looking into this and I do not think that is an issue with Robomongo.  We just needed to grant the `clusterMonitor` role to the user and now it can see all the databases in the tree view.

```javascript
db.grantRolesToUser(
    "joe_user",
    [
      { role: "clusterMonitor", db: "admin" }
    ]
)
```

**Comment by goldiiman** [comment]
My installation is locked down so i dont want a user to see anything but
what they are allowed to.


On 6 Sep 2017 13:57, "Chuck May" <notifications@github.com> wrote:

Did some more looking into this and I do not think that is an issue with
Robomongo. We just needed to grant the clusterMonitor role to the user and
now it can see all the databases in the tree view.

db.grantRolesToUser(
    "joe_user",
    [
      { role: "clusterMonitor", db: "admin" }
    ]
)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Studio3T/robomongo/issues/1357#issuecomment-327474819>,
or mute the thread

**Comment by ctmay4** [comment]
That's a good point, but they would only see the name of the database in the list.  They would not have access to see the collection names, etc.

**Comment by goldiiman** [comment]
Yes but in my case these database names are auto generated per user so
there would be 100s where they only have access to one.



On 6 Sep 2017 14:01, "Chuck May" <notifications@github.com> wrote:

That's a good point, but they would only see the name of the database in
the list. They would not have access to see the collection names, etc.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Studio3T/robomongo/issues/1357#issuecomment-327475861>,
or mute the thread
